templates/restful_service/src/restful_service/main.bal

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/config;
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+
+// The stock quote management is done using the SimpleStockQuoteService.
+// Define a listener endpoint:
+listener http:Listener httpListener = new(config:getAsInt("LISTENER_PORT"));
+
+// Constants.
+string STOCK_QUOTE_SERVICE_BASE_URL = config:getAsString("STOCK_QUOTE_SERVICE_URL", "http://localhost:9000");
+const string ERROR_MESSAGE_WHEN_RESPOND = "Error while sending response to the client";
+const string ERROR_MESSAGE_INVALID_PAYLOAD = "Invalid payload received";
+
+http:Client stockQuoteClient = new(STOCK_QUOTE_SERVICE_BASE_URL);
+
+// RESTful service.
+@http:ServiceConfig { basePath: "/stockQuote" }
+service stockQuote on httpListener {
+
+    // Resource that handles the HTTP GET requests that are directed to a specific stock using path '/quote/<symbol>'.
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/quote/{symbol}"
+    }
+    resource function getQuote(http:Caller caller, http:Request request, string symbol) {
+        // Find the quote details of a single stock. The response contains the last sale price of the stock.
+        xml payload = xml `<m0:getQuote xmlns:m0="http://services.samples">
+                               <m0:request>
+                                   <m0:symbol>${symbol}</m0:symbol>
+                               </m0:request>
+                           </m0:getQuote>`;
+        xml soapEnv = self.constructSOAPPayload(<@untainted> payload, "http://schemas.xmlsoap.org/soap/envelope/");
+
+        request.addHeader("SOAPAction", "urn:getQuote");
```

Comment:
Can we add some comment for the following three lines of code?

Reply:
Added coments